### ISSUE TITLE
Remove all warnings during compilation

### ISSUE THREAD

**Issue by blattms** [OPM/opm-common#723]
Even if PR #722 is merged.there will be the following warnings left:

```

/home/mblatt/src/dune/opm/opm-common/examples/test_util/EclFile.cpp:430:26: warning: ‘std::vector<std::__cxx11::basic_string<char> > {anonymous}::readBinaryCharArray(std::fstream&, int)’ defined but not used [-Wunused-function]
 std::vector<std::string> readBinaryCharArray(std::fstream &fileH, const int size){
                          ^~~~~~~~~~~~~~~~~~~
/home/mblatt/src/dune/opm/opm-common/examples/test_util/EclFile.cpp:359:19: warning: ‘std::vector<bool> {anonymous}::readBinaryLogiArray(std::fstream&, int)’ defined but not used [-Wunused-function]
 std::vector<bool> readBinaryLogiArray(std::fstream &fileH, const int size){
                   ^~~~~~~~~~~~~~~~~~~
/home/mblatt/src/dune/opm/opm-common/examples/test_util/EclFile.cpp:301:21: warning: ‘std::vector<double> {anonymous}::readBinaryDoubArray(std::fstream&, int)’ defined but not used [-Wunused-function]
 std::vector<double> readBinaryDoubArray(std::fstream &fileH, const int size){
                     ^~~~~~~~~~~~~~~~~~~
/home/mblatt/src/dune/opm/opm-common/examples/test_util/EclFile.cpp:242:20: warning: ‘std::vector<float> {anonymous}::readBinaryRealArray(std::fstream&, int)’ defined but not used [-Wunused-function]
 std::vector<float> readBinaryRealArray(std::fstream &fileH, const int size){
                    ^~~~~~~~~~~~~~~~~~~
/home/mblatt/src/dune/opm/opm-common/examples/test_util/EclFile.cpp:184:18: warning: ‘std::vector<int> {anonymous}::readBinaryInteArray(std::fstream&, int)’ defined but not used [-Wunused-function]
 std::vector<int> readBinaryInteArray(std::fstream &fileH, const int size){
                  ^~~~~~~~~~~~~~~~~~~
/home/mblatt/src/dune/opm/opm-common/examples/test_util/EclFile.cpp:155:19: warning: ‘long unsigned int {anonymous}::sizeOnDisk(int, const Opm::ecl::eclArrType&)’ defined but not used [-Wunused-function]
 unsigned long int sizeOnDisk(const int num, const EIOD::eclArrType &arrType){
                   ^~~~~~~~~~
/home/mblatt/src/dune/opm/opm-common/examples/test_util/EclFile.cpp:101:6: warning: ‘void {anonymous}::readBinaryHeader(std::fstream&, std::__cxx11::string&, int&, Opm::ecl::eclArrType&)’ defined but not used [-Wunused-function]
 void readBinaryHeader(std::fstream &fileH, std::string &arrName, int &size, EIOD::eclArrType &arrType){
      ^~~~~~~~~~~~~~~~
/home/mblatt/src/dune/opm/opm-common/examples/test_util/EclFile.cpp:33:6: warning: ‘bool {anonymous}::isEOF(std::fstream*)’ defined but not used [-Wunused-function]
 bool isEOF(std::fstream *fileH){
      ^~~~~
```
and
```
/home/mblatt/src/dune/opm/opm-common/src/opm/output/eclipse/libECLRestart.cpp: In function ‘void Opm::RestartIO::ecl_kw_set_data_type(Opm::RestartIO::ecl_kw_type*, Opm::RestartIO::ecl_data_type)’:
/home/mblatt/src/dune/opm/opm-common/src/opm/output/eclipse/libECLRestart.cpp:720:60: warning: ‘void* memcpy(void*, const void*, size_t)’ writing to an object of type ‘Opm::RestartIO::ecl_data_type’ {aka ‘struct Opm::RestartIO::ecl_type_struct’} with no trivial copy-assignment; use copy-initialization instead [-Wclass-memaccess]
     memcpy(&ecl_kw->data_type, &data_type, sizeof data_type);
                                                            ^
In file included from /home/mblatt/src/dune/opm/opm-common/src/opm/output/eclipse/libECLRestart.cpp:23:
/home/mblatt/src/dune/opm/opm-common/opm/output/eclipse/libECLRestart.hpp:350:8: note: ‘Opm::RestartIO::ecl_data_type’ {aka ‘struct Opm::RestartIO::ecl_type_struct’} declared here
 struct ecl_type_struct {
        ^~~~~~~~~~~~~~~
/home/mblatt/src/dune/opm/opm-common/src/opm/output/eclipse/libECLRestart.cpp: In function ‘Opm::RestartIO::ecl_file_kw_type* Opm::RestartIO::ecl_file_kw_alloc0(const char*, Opm::RestartIO::ecl_data_type, int, Opm::RestartIO::offset_type)’:
/home/mblatt/src/dune/opm/opm-common/src/opm/output/eclipse/libECLRestart.cpp:3230:59: warning: ‘void* memcpy(void*, const void*, size_t)’ writing to an object of type ‘Opm::RestartIO::ecl_data_type’ {aka ‘struct Opm::RestartIO::ecl_type_struct’} with no trivial copy-assignment; use copy-initialization instead [-Wclass-memaccess]
   memcpy(&file_kw->data_type, &data_type, sizeof data_type);
                                                           ^
In file included from /home/mblatt/src/dune/opm/opm-common/src/opm/output/eclipse/libECLRestart.cpp:23:
/home/mblatt/src/dune/opm/opm-common/opm/output/eclipse/libECLRestart.hpp:350:8: note: ‘Opm::RestartIO::ecl_data_type’ {aka ‘struct Opm::RestartIO::ecl_type_struct’} declared here
 struct ecl_type_struct {
        ^~~~~~~~~~~~~~~
```
Maybe some more knowlegable could take a look here.

**Comment by bska** [comment]
Anything in in `libECLRestart.*` will go away shortly after #703 is merged.  I'm working on making the restart code use the facilities introduced in that PR at which point we'll be able to drop the `libECLRestart` component.

**Comment by bska** [comment]
Just a quick follow-up: Do you still get warnings about functions that are `defined but not used` in [`EclFile.cpp`](https://github.com/OPM/opm-common/blob/bf6964522e5a7a756abd038aa175d486d28afb52/examples/test_util/EclFile.cpp), even after PR #699 was merged into master?

**Comment by blattms** [comment]
Closing as won't fix.
